backend/api/quivr_api/modules/sync/utils/syncutils.py

```diff
@@ -26,73 +23,47 @@
 from quivr_api.modules.sync.entity.sync import SyncFile
 from quivr_api.modules.sync.repository.sync_files import SyncFiles
 from quivr_api.modules.sync.service.sync_service import SyncService, SyncUserService
-from quivr_api.modules.sync.utils.list_files import (
-    get_azure_files_by_id,
-    list_azure_files,
-)
+from quivr_api.modules.sync.utils.sync import BaseSync
 from quivr_api.modules.sync.utils.upload import upload_file
 from quivr_api.modules.upload.service.upload_file import check_file_exists
 
 notification_service = NotificationService()
-
 logger = get_logger(__name__)
 
-CLIENT_ID = os.getenv("SHAREPOINT_CLIENT_ID")
-AUTHORITY = "https://login.microsoftonline.com/common"
-BACKEND_URL = os.getenv("BACKEND_URL", "http://localhost:5050")
-REDIRECT_URI = f"{BACKEND_URL}/sync/azure/oauth2callback"
-SCOPE = [
-    "https://graph.microsoft.com/Files.Read",
-    "https://graph.microsoft.com/User.Read",
-    "https://graph.microsoft.com/Sites.Read.All",
-]
-
 
-class AzureSyncUtils(BaseModel):
+class SyncUtils(BaseModel):
     model_config = ConfigDict(arbitrary_types_allowed=True)
 
     sync_user_service: SyncUserService
     sync_active_service: SyncService
     sync_files_repo: SyncFiles
     storage: Storage
-
-    def get_headers(self, token_data):
-        return {
-            "Authorization": f"Bearer {token_data['access_token']}",
-            "Accept": "application/json",
-        }
-
-    def refresh_token(self, refresh_token):
-        client = msal.PublicClientApplication(CLIENT_ID, authority=AUTHORITY)
-        result = client.acquire_token_by_refresh_token(refresh_token, scopes=SCOPE)
-        if "access_token" not in result:
-            raise HTTPException(status_code=400, detail="Failed to refresh token")
-        return result
+    sync_cloud: BaseSync
 
     async def _upload_files(
```

Comment:
Looking at the code. This function probably needs to be refactored. We could create a refacto ticket for this and for later.

Reply:
Yes I'm creating the ticket !